patches/server/1062-Add-ShulkerDuplicateEvent.patch

```diff
@@ -0,0 +1,23 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Chase Henderson <henderson.chase@gmail.com>
+Date: Fri, 5 Jan 2024 03:50:10 -0500
+Subject: [PATCH] Add ShulkerDuplicateEvent
+
+
+diff --git a/src/main/java/net/minecraft/world/entity/monster/Shulker.java b/src/main/java/net/minecraft/world/entity/monster/Shulker.java
+index b73dac8f68041f8a2e0752d70cc9d08b5cfd1cde..e02c9c9abbde9b18fad1398d3096961430e21810 100644
+--- a/src/main/java/net/minecraft/world/entity/monster/Shulker.java
++++ b/src/main/java/net/minecraft/world/entity/monster/Shulker.java
+@@ -484,6 +484,12 @@ public class Shulker extends AbstractGolem implements VariantHolder<Optional<Dye
+                 if (entityshulker != null) {
+                     entityshulker.setVariant(this.getVariant());
+                     entityshulker.moveTo(vec3d);
++                    // Paper start - Shulker duplicate event
++                    if (!new io.papermc.paper.event.entity.ShulkerDuplicateEvent((org.bukkit.entity.Shulker) entityshulker.getBukkitEntity(), (org.bukkit.entity.Shulker) this.getBukkitEntity()).callEvent()) {
++                        entityshulker.discard();
```

Comment:
This is interesting to me... Does this happen elsewhere for events relating to the spawning of entities? If it does, then you can leave the discard, I just don't remember exactly how the handling of this is done elsewhere.

Reply:
That was why originally I did it after the add fresh entity. I could not find any other examples similar to this scenario. Not discarding it seemed wrong since the entity would remain unmarked yet also unadded to the world.

Reply:
`EntityBreedEvent` has similar logic, but does not have a discard.

Reply:
Thanks for some reason didn't see entity breed event had the child.